### PR TITLE
feat: set useSemicolonDefault default option to false

### PR DESCRIPTION
**Pull request by dancastillo**
#### Checklist

closes: https://github.com/fastify/fastify/issues/5319

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Uzlopak** [comment]
We should somehow track those changes so that we dont have the struggle to create a migration guide in few months.

**Review by gurgunday — APPROVED** [review]

**Comment by mcollina** [comment]
Can you start such a document, @dancastillo? Something like https://github.com/fastify/fastify/blob/main/docs/Guides/Migration-Guide-V4.md.

**Comment by dancastillo** [comment]
@Uzlopak @mcollina added in the migration guide for v5 [here](https://github.com/fastify/fastify/pull/5320/files#diff-44f6559b13d8fc083ee5bae265ce8f14cc2fe45e26f8d0ad04fac69b75b3ea90)

**Review by metcoder95 — APPROVED** [review]

File: docs/Guides/Migration-Guide-V5.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+Before migrating to v5, please ensure that you have fixed all deprecation
+warnings from v4. All v4 deprecations have been removed and they will no longer
+work after upgrading.
+
+## Breaking Changes
+
+### `useSemicolonDelimiter` false by default
+
+Starting with v5, every Fastify instance will not support the use of semicolon
+delimiter by default. You can revert this behavior by setting
```
Comment (mcollina) [inline docs/Guides/Migration-Guide-V5.md:14]:
```suggestion
delimiter by default in querystrings. You can revert this behavior by setting
```

**Comment by Uzlopak** [comment]
I think we can be a little bit more opinionated and inform the reader that this behaviour was always non standard behaviour and it should be having no effect on the standard implementations.

**Comment by jsumners** [comment]
> We should somehow track those changes so that we dont have the struggle to create a migration guide in few months.

https://github.com/orgs/fastify/projects/8

File: docs/Guides/Migration-Guide-V5.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+## Breaking Changes
+
+### `useSemicolonDelimiter` false by default
+
+Starting with v5, every Fastify instance will not support the use of semicolon
+delimiter by default in querystring. The behavior in v4 and eariler versions is
+non standard behavior and this change should have no effect on the standard
+implementation of Fastify. You can revert this behavior by setting
+`useSemicolonDelimiter: true` in the server options.
```
Comment (Uzlopak) [inline docs/Guides/Migration-Guide-V5.md:17]:
```suggestion
Beginning with version 5, Fastify instances will no longer utilize semicolons as delimiters when parsing querystrings by default. This decision was made due to the non-standard nature of semicolon delimiters, which are not supported by widely used querystring parsers like the built-in querystring package in Node.js.

However, if you still wish to use semicolons as delimiters, you can do so by setting the useSemicolonDelimiter option to true in the server configuration.
```

**Comment by Uzlopak** [comment]
@jsumners 

Sure, but it is better to formulate the migration guide now then in few weeks where we maybe forgot some edge cases etc..

@dancastillo 
Thank you for your work. I proposed an imho better description with the help of chatGPT :P

image

File: docs/Guides/Migration-Guide-V5.md (hunk 2 above)
Comment (Uzlopak) [reply docs/Guides/Migration-Guide-V5.md:17]: Before you merge, @Fdawgs  and @jsumners you ok with the rephrasing?

Comment (Fdawgs) [reply docs/Guides/Migration-Guide-V5.md:17]: @Uzlopak i can take a look tomorrow

Comment (Uzlopak) [reply docs/Guides/Migration-Guide-V5.md:17]: @jsumners 
thanks bro :)

**Review by Uzlopak — APPROVED** [review]

**Pull request merged**